### PR TITLE
Feature/typeahead

### PR DESCRIPTION
**Pull request by CloudNiner**
For now, the typeahead always loads the map state with the params changed. In the upcoming UI changes, there will be a typeahead on both the home page and map pages and they will work slightly differently.

### CODE REVIEW

File: app/scripts/geocoder/geocoder-service.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        var suggestUrl = 'http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/suggest';
+        var boundingBox = [
+            Config.bounds.southWest.lng,
+            Config.bounds.southWest.lat,
+            Config.bounds.northEast.lng,
+            Config.bounds.northEast.lat
+        ].join(',');
+        var maxResults = 10;
+        var searchOutFields = 'StAddr,City,Postal';
+        var searchCategories = [
```
Comment (flibbertigibbet) [inline app/scripts/geocoder/geocoder-service.js:19]: No place names?

**Comment by flibbertigibbet** [comment]
+1

File: app/scripts/geocoder/geocoder-service.js (hunk 1 above)
Comment (CloudNiner) [inline app/scripts/geocoder/geocoder-service.js:19]: Nope, just addresses and neighborhoods according to @jbranigan

Comment (flibbertigibbet) [inline app/scripts/geocoder/geocoder-service.js:19]: Well that makes things easier.

**Pull request merged**